packages/visual-editor/README.md

```diff
@@ -1,7 +1,23 @@
+### Purpose
+- Handles the Canvas Interaction logic with dragging, hitboxes, dropzones, reparenting, and communicating with the parent frame to ensure a smooth drag and drop experience.
+- Note that this package is framework specific to React.
 
-## Documentation
+### Concepts
+- **Structure Components**: Represents components that pertain to the layout of a user's experience such as containers, sections, and columns.
+- **Built-In Components**: Contentful provided components on behalf of the user that are considered universally useful such as an image, text, and the divider component or to get a user quickly started such as a button. Note that these components are not meant to cover every possible component that the user may want to have. Instead users are generally expected to provide their own set of components to build their customized experiences. The exceptions will generally be based on how universally desired such components are, such as Cards or Carousel components in the future.
+- **Custom Components**: User provided components built through a definition provided by the Experiences sdk.
+- **Post messages**: In order for an iFrame to communicate with its parent frame, post messaging is required so that the two frames can communicate with each other. Examples of what information gets shared would be mouse coordinates or what component got picked up and dropped.
+- **Browser Limitations**: Currently only Chrome supports both Click/Drag and Drop while Safari and Firefox only allows for Click and Drop.
+- **Trees**: Experiences store their own version of the tree that is separate from the source of truth which is stored as a Contentful entry. This is because by storing its own tree version it can provide ui optimizations and manipulations to the tree to maximize performance related tasks such as when performing drag and drop or re-ordering of components. A concrete example where this optimization is necessary is when you drop a component onto the canvas. Without its own version, the component would flicker to its original pre-drag spot before being placed on the intended area based on the mouse's location. 
+- **Updating the source of truth**: On drag end, the iFrame will update its own tree before sending a post message to the parent frame in `useCanvasInteractions`. Then the parent frame will update its tree which is the source of truth before sending a post msg back in  `useEditorSubscriber` to the iFrame ensuring that both trees are the same and any discovered discrepancies are resolved.
+- **Pangea DND**: Library providing HOC (higher order components) that Experiences leverages for Drag and Drop. The two components being used from this library is the `Dropzone` and `Draggable` component.
```

Comment:
"Pangea DND" seems like slang to me. Should we mention the library name as `@hello-pangea/dnd`?

Reply:
> The two components being used from this library is the `Dropzone` and `Draggable` component.

We're using `Draggable` and `Droppable` from this library, and `Dropzone` is a custom component.

```suggestion
- **@hello-pangea/dnd**: Library providing drag and drop capabilities for the Experiences Canvas. The two components being used from this library are the `Draggable` and `Droppable` components.
```